minitwit/src/main/java/WebApplication.java

```diff
@@ -556,4 +570,22 @@ else if (!BCrypt.checkpw(enteredPW, rs.getString("pw_hash"))) {
         });
         return filteredMessages.toList();
     };
+
+    public static Route serveSimRegister = (Request request, Response response) -> {
+        var json = gson.fromJson(request.body(), HashMap.class);
+        var username = String.valueOf(json.get("username"));
+        var email = String.valueOf(json.get("email"));
+        var password = String.valueOf(json.get("pwd"));
+
+        var error = register(username, email, password, password);
+        if (error != null) {
+            response.status(400);
+            return Map.ofEntries(
+                    Map.entry("status", 400),
+                    Map.entry("error_msg", error)
+            );
+        }
+
+        return Map.ofEntries();
```

Comment:
The spec just returns a empty body but I couldn't figure out a neat way to do that with out gson setup, so now we return a empty object as body...

Reply:
I think that's fine, but looks like the Python method returns 204 as a status code as well as the empty body. Maybe we need that here too? Not sure where we specify status codes in our program.

Reply:
Reading the simulators code it looks like the exact return codes are important so I'll try figure something out.